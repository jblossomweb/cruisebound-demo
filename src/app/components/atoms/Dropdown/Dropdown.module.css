.select {
  @apply font-normal
    text-base
    text-gray-500
    p-0
    cursor-pointer
    border
    border-gray-300
    rounded
    inline-block;
}

.select.wide {
  @apply w-full;
}

.select.naked .inner {
  -moz-appearance: none;
  -webkit-appearance: none;
}

.select.shadow {
  @apply drop-shadow-lg;
}

.label {
  @apply block pb-2;
}

.select .inner {
  @apply p-2
    border-transparent
    border-r-8
    w-full
    rounded
    cursor-pointer;
}
