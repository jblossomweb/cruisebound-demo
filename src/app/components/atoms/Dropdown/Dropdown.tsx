import React from 'react'
import classNames from 'classnames'

import type SelectOption from 'app/types/SelectOption'
import styles from './Dropdown.module.css'

export interface DropdownProps {
  label?: string
  value?: SelectOption['value']
  options: SelectOption[]
  wide?: boolean
  naked?: boolean
  shadow?: boolean
  // eslint-disable-next-line no-unused-vars
  onSelect: (value: SelectOption['value']) => void
}

const Dropdown: React.FC<DropdownProps> = ({
  label = undefined,
  value = undefined,
  options,
  wide = false,
  naked = false,
  shadow = false,
  onSelect,
}) => (
  <>
    {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
    {!!label && <label className={styles.label}>{label}</label>}
    <div
      className={classNames(styles.select, {
        [styles.wide]: wide,
        [styles.naked]: naked,
        [styles.shadow]: shadow,
      })}
    >
      <select
        data-testid="select"
        className={styles.inner}
        value={value}
        onChange={(e) => {
          onSelect(e.target.value)
        }}
      >
        {options.map((option) => (
          <option key={option.value} value={option.value}>
            {option.display}
          </option>
        ))}
      </select>
    </div>
  </>
)

export default Dropdown
