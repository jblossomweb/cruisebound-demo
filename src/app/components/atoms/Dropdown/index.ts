export type { ButtonProps } from './Dropdown'
export { default } from './Dropdown'
