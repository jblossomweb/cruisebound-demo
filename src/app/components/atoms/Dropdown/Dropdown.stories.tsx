import React from 'react'
import type { StoryFn, Meta } from '@storybook/react'

import Dropdown, { type DropdownProps } from './Dropdown'

export default {
  title: 'components/atoms/Dropdown',
  component: Dropdown,
  argTypes: {
    options: { control: 'object' },
    label: { control: 'text' },
    value: { control: 'text' },
    naked: { control: 'boolean' },
    shadow: { control: 'boolean' },
    wide: { control: 'boolean' },
    onSelect: { control: 'none' },
  },
} as Meta<typeof Dropdown>

const Template: StoryFn<typeof Dropdown> = (args: DropdownProps) => (
  <Dropdown {...args} />
)

export const Sample = Template.bind({})
Sample.args = {
  options: [
    { display: 'Foo', value: 'foo' },
    { display: 'Bar', value: 'bar' },
    { display: 'Buzz', value: 'buzz' },
  ],
}

export const WithLabel = Template.bind({})
WithLabel.args = {
  ...Sample.args,
  label: 'label',
}

export const Naked = Template.bind({})
Naked.args = {
  ...Sample.args,
  naked: true,
}

export const Shadow = Template.bind({})
Shadow.args = {
  ...Sample.args,
  shadow: true,
}

export const Wide = Template.bind({})
Wide.args = {
  ...Sample.args,
  wide: true,
}
