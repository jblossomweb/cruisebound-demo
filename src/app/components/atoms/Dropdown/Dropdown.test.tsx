/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import storySnaps from 'test/utils/storySnaps'

import Dropdown from './Dropdown'
import * as stories from './Dropdown.stories'

describe('app/components/atoms/Dropdown', () => {
  describe('stories', () => {
    storySnaps(stories)
  })

  describe('interaction', () => {
    afterEach(cleanup)

    it('should call onSelect when selection changes', () => {
      const options = [
        { display: 'Foo', value: 'foo' },
        { display: 'Bar', value: 'bar' },
        { display: 'Buzz', value: 'buzz' },
      ]
      const onSelect = jest.fn()
      expect(onSelect).not.toHaveBeenCalled()
      render(<Dropdown options={options} onSelect={onSelect} />)
      expect(onSelect).not.toHaveBeenCalled()
      const select = screen.getByTestId('select')
      fireEvent.change(select, { target: { value: 'buzz' } })
      expect(onSelect).toHaveBeenCalled()
      expect(onSelect).toHaveBeenCalledWith('buzz')
    })
  })
})
