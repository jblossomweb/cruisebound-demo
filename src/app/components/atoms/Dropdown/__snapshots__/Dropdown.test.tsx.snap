// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/atoms/Dropdown stories Naked matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="select naked"
  >
    <select
      class="inner"
      data-testid="select"
    >
      <option
        value="foo"
      >
        Foo
      </option>
      <option
        value="bar"
      >
        Bar
      </option>
      <option
        value="buzz"
      >
        Buzz
      </option>
    </select>
  </div>
</DocumentFragment>
`;

exports[`app/components/atoms/Dropdown stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="select"
  >
    <select
      class="inner"
      data-testid="select"
    >
      <option
        value="foo"
      >
        Foo
      </option>
      <option
        value="bar"
      >
        Bar
      </option>
      <option
        value="buzz"
      >
        Buzz
      </option>
    </select>
  </div>
</DocumentFragment>
`;

exports[`app/components/atoms/Dropdown stories Shadow matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="select shadow"
  >
    <select
      class="inner"
      data-testid="select"
    >
      <option
        value="foo"
      >
        Foo
      </option>
      <option
        value="bar"
      >
        Bar
      </option>
      <option
        value="buzz"
      >
        Buzz
      </option>
    </select>
  </div>
</DocumentFragment>
`;

exports[`app/components/atoms/Dropdown stories Wide matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="select wide"
  >
    <select
      class="inner"
      data-testid="select"
    >
      <option
        value="foo"
      >
        Foo
      </option>
      <option
        value="bar"
      >
        Bar
      </option>
      <option
        value="buzz"
      >
        Buzz
      </option>
    </select>
  </div>
</DocumentFragment>
`;

exports[`app/components/atoms/Dropdown stories WithLabel matches markup snapshot 1`] = `
<DocumentFragment>
  <label
    class="label"
  >
    label
  </label>
  <div
    class="select"
  >
    <select
      class="inner"
      data-testid="select"
    >
      <option
        value="foo"
      >
        Foo
      </option>
      <option
        value="bar"
      >
        Bar
      </option>
      <option
        value="buzz"
      >
        Buzz
      </option>
    </select>
  </div>
</DocumentFragment>
`;
